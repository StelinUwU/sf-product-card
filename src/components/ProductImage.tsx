import React, { useContext } from 'react';
import { ProductContext } from './ProductCard';
import noImage from '../assets/no-image.jpg';
import styles from '../styles/styles.module.css';

export interface Props {
  className?: string;
  img?: string;
  style?: React.CSSProperties;
}

export const ProductImage = ({ img, className, style }: Props) => {
  const { product } = useContext(ProductContext);
  const image = img || product.img || noImage;
  return (
    <img
      className={`${styles.productImg} ${className}`}
      src={image}
      style={style}
      alt="Product img"
    />
  );
};
