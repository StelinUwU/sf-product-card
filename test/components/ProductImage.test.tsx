import React from 'react';
import renderer from 'react-test-renderer';
import { ProductImage, ProductCard } from '../../src/components';
import { product2 } from '../data/products';

describe('ProductTitle', () => {
  test('should render the component according to the props', () => {
    const wrapper = renderer.create(
      <ProductImage img={product2.img} className="custom-class" />
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('should render the component with the product image', () => {
    const wrapper = renderer.create(
      <ProductCard product={product2}>{() => <ProductImage />}</ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
