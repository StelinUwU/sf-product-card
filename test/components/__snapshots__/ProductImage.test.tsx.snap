// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle should render the component according to the props 1`] = `
<img
  alt="Product img"
  className="productImg custom-class"
  src="./coffee-mug.png"
/>
`;

exports[`ProductTitle should render the component with the product image 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product img"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;
